packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,683 @@
+      for(int i = 0; i < widget.children.length; i += 1) {
+        if(widget.children.elementAt(i).key == widget.value) {
+          _selectedValue = ValueKey<int>(i);
+        }
+      }
+    }
+
+  @override
+  Widget build(BuildContext context) {
+    _gestureChildren.clear();
+    for (int i = 0; i < widget.children.length; i += 1) {
+      final Key childKey = ValueKey<int>(i);
+      final TextStyle textStyle = DefaultTextStyle.of(context).style.copyWith(
+            color: (_selectedValue == childKey)
+                ? CupertinoColors.white
+                : CupertinoColors.activeBlue,
+          );
+      final IconThemeData iconTheme = IconThemeData(
+        color: (_selectedValue == childKey)
+            ? CupertinoColors.white
+            : CupertinoColors.activeBlue,
+      );
+
+      Widget child = widget.children.elementAt(i);
+      child = GestureDetector(
+        onTapDown: (TapDownDetails event) {
+          _onTapDown(childKey);
+        },
+        onTapUp: (TapUpDetails event) {
+          _onTapUp();
+        },
+        onTapCancel: () {
+          _onTapCancel();
+        },
+        onTap: () {
+          _onTap(widget.children.elementAt(i).key, childKey);
+        },
+        child: new IconTheme(
+            data: iconTheme,
+            child: DefaultTextStyle(
+              style: textStyle,
+              overflow: TextOverflow.ellipsis,
+              child: Semantics(
+                  inMutuallyExclusiveGroup: true,
+                  selected: _selectedValue != null
+                      ? _selectedValue.value == i
+                      : false,
+                  child: child),
+            )),
+      );
+      _gestureChildren.add(child);
+    }
+
+    final Widget box = new _SegmentedControlRenderWidget(
+      children: _gestureChildren,
+      selectedValue: _selectedValue,
+      pressedValue: _pressedValue,
+    );
+
+    return Padding(
+      padding: _kHorizontalItemPadding.resolve(Directionality.of(context)),
+      child: UnconstrainedBox(
+        constrainedAxis: Axis.horizontal,
+        child: box,
+      ),
+    );
+  }
+}
+
+class _SegmentedControlRenderWidget extends RenderObjectWidget {
+  const _SegmentedControlRenderWidget({
+    Key key,
+    @required this.children,
+    @required this.selectedValue,
+    @required this.pressedValue,
+  });
+
+  final List<Widget> children;
+  final ValueKey<int> selectedValue;
+  final ValueKey<int> pressedValue;
+
+  @override
+  RenderObjectElement createElement() => new _RenderSegmentedControlElement(this);
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return new _RenderSegmentedControl(
+      textDirection: Directionality.of(context),
+      children: children,
+      selectedValue: selectedValue,
+      pressedValue: pressedValue,
+    );
+  }
+
+  @override
+  void updateRenderObject(
+      BuildContext context, _RenderSegmentedControl renderObject) {
+    renderObject.markNeedsPaint();
```

Comment:
this is wrong and should not be necessary. the setters should take care of this automatically.